### PR TITLE
GPU Operator: add no_undeploy flag support and various updates

### PR DESCRIPTION
**Pull request by kpouget**
* `CI: GPU Operator: add no_undeploy flag support`

To avoid failing during upgrade testing

---

* `CI: GPU Operator: remove helm entrypoint`

---

* `CI: GPU Operator: allow 'sourcing' this file`

---

* `gpu_operator_capture_deployment_state: never fail`

---

* `cluster_upgrade_to_image: give 120min for the upgrade to complete`

---

* `capture_environment: capture the clusterversion/version`

---

* `toolbox: gpu-operator/diagnose.sh: reorder the artifacts directories`

---

* `gpu_operator_capture_deployment_state: improve the capture`

---

* `capture_environment: rename into cluster_capture_environment`

---

* `toolbox/_common.py: use os.execvpe instead of subprocess.run`

---

* `CI: run: git show HEAD and all of its parent commits`

---

### CODE REVIEW

**Comment by kpouget** [comment]
goes along with https://github.com/openshift/release/pull/21730

**Comment by kpouget** [comment]
```
2021-09-10 06:55:45,208 p=2465 u=psap-ci-runner n=ansible | ----- FAILED ----
2021-09-10 06:55:45,208 p=2465 u=psap-ci-runner n=ansible | msg: non-zero return code
2021-09-10 06:55:45,208 p=2465 u=psap-ci-runner n=ansible | 
2021-09-10 06:55:45,208 p=2465 u=psap-ci-runner n=ansible | <command> oc get packagemanifests/nfd -n openshift-marketplace -ojsonpath={.status.defaultChannel}
2021-09-10 06:55:45,208 p=2465 u=psap-ci-runner n=ansible | 
2021-09-10 06:55:45,208 p=2465 u=psap-ci-runner n=ansible | <stderr> Error from server (NotFound): packagemanifests.packages.operators.coreos.com "nfd" not found
2021-09-10 06:55:45,209 p=2465 u=psap-ci-runner n=ansible | ----- FAILED ----
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/254/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-e2e/1436218042789924864/artifacts/gpu-operator-e2e/presubmit-operatorhub/artifacts/005__nfd_operator__deploy_from_operatorhub/_ansible.log

that didn't happen for quite a while!

/test gpu-operator-e2e

**Comment by kpouget** [comment]
>  ci/prow/4.6-gpu-operator-e2e — Job failed.                    
```
* could not run steps: step gpu-operator-e2e failed: failed to wait for the created cluster claim to become ready: timed out waiting for the condition 
```
:( 

trying again, but maybe we're requestiong 4.6 clusters from a non-existing pool (or pool is just empty, we'll see)

File: toolbox/_common.py [hunk 1]
```diff
@@ -107,4 +106,6 @@ def run_ansible_playbook(playbook_name, opts: dict = dict()):
         for k, v in os.environ.items():
             print(f"{k}={v}", file=f)
 
-    raise SystemExit(run_result.returncode)
+    os.execvpe(cmd[0], cmd, os.environ)
```
Comment (omertuc) [inline toolbox/_common.py:109]: Isn't this already a part of another pull request?

Comment (kpouget) [reply toolbox/_common.py:109]: this commit was part of https://github.com/openshift-psap/ci-artifacts/pull/235 but I moved it here, along with the "various updates" of the PR title

**Comment by kpouget** [comment]
> /lgtm

thanks @omertuc, let's see if we can get green testing today

/test gpu-operator-e2e

**Comment by kpouget** [comment]
@omertuc `ci/prow/4.9-gpu-operator-e2e` test is passing, good for merging

/approve

**Comment by kpouget** [comment]
@omertuc good for merging?

**Comment by kpouget** [comment]
@dagrayvid any feedback on this PR?

**Pull request merged**